Makefile

```diff
@@ -59,26 +62,28 @@ endif
 # To update upstream dependencies, delete the glide.lock file first.
 vendor: glide.yaml
 	# To build without Docker just run "glide install -strip-vendor"
-	if [ "$(LIBCALICOGO_PATH)" != "none" ]; then \
-          EXTRA_DOCKER_BIND="-v $(LIBCALICOGO_PATH):/go/src/github.com/projectcalico/libcalico-go:ro"; \
+	-mkdir -p ~/.glide
+	if [ "$(LIBCALICOGO_PATH)" != "none"  ]; then \
+	  EXTRA_DOCKER_BIND="-v $(LIBCALICOGO_PATH):/go/src/github.com/projectcalico/libcalico-go:ro"; \
 	fi; \
 	docker run --rm \
-	-v ${PWD}:/go/src/github.com/projectcalico/cni-plugin:rw $$EXTRA_DOCKER_BIND \
-      --entrypoint /bin/sh $(GO_CONTAINER_NAME) -e -c ' \
-	cd /go/src/github.com/projectcalico/cni-plugin && \
-	glide install -strip-vendor && \
-	chown $(shell id -u):$(shell id -u) -R vendor'
+	  -v $(CURDIR):/go/src/github.com/projectcalico/cni-plugin:rw $$EXTRA_DOCKER_BIND \
+		-v $(HOME)/.glide:/home/user/.glide:rw \
+		-e LOCAL_USER_ID=$(LOCAL_USER_ID) \
+		$(CALICO_BUILD) /bin/sh -c ' \
+			cd /go/src/github.com/projectcalico/cni-plugin && \
+			glide install -strip-vendor' 
```

Comment:
I think strip vendor _is_ still needed. ANd one dash works just fine (but I can update it to two)

```
docker run --rm calico/go-build glide install --help 
Starting with UID : 9001
NAME:
   glide install - Install a project's dependencies

USAGE:
   glide install [command options] [arguments...]

DESCRIPTION:
   This uses the native VCS of each packages to install
   the appropriate version. There are two ways a projects dependencies can
   be installed. When there is a glide.yaml file defining the dependencies but
   no lock file (glide.lock) the dependencies are installed using the "update"
   command and a glide.lock file is generated pinning all dependencies. If a
   glide.lock file is already present the dependencies are installed or updated
   from the lock file.

OPTIONS:
   --force             If there was a change in the repo or VCS switch to new one. Warning: changes will be lost.
   --strip-vendor, -v  Removes nested vendor and Godeps/_workspace directories.
   --skip-test         Resolve dependencies in test files.
```